### ISSUE TITLE
Awaiting for initialisation of next-i18next instance

### ISSUE THREAD

**Issue by carloscuesta** [isaachinman/next-i18next#564]
Hi @isaachinman 👋 

Referring to the next issue https://github.com/isaachinman/next-i18next/issues/374#issuecomment-529196313 specifically this comment:

> Finally took some time to dig into this. The issue of awaiting the initialisation of the i18next instance is a very trivial one, and I can provide an easy, non-breaking way for users to do that if they want to. However, I think that that race condition is causing a very low percentage of the error reports we are seeing here.

Would you mind on sharing the approach of awaiting the initialisation of `next-i18next` instance, please ?

We're having some issues with it and we haven't found a way to await it or initialise it synchronously 

Thanks 🙏

**Comment by isaachinman** [comment]
Hi @carloscuesta. So, after doing some stuff with user options and plugins, etc, `next-i18next` ultimately [just calls `i18n.init`](https://github.com/isaachinman/next-i18next/blob/88033719baf46dd7bc0ca02f0cc35b0e98bce8af/src/create-i18next-client.ts#L28).

The `.init` function returns a promise, and I believe there's also some sort of callback option as well.

There is already a boolean flag on the i18n instance called `isInitialized`, which you could observe and await.

Or, we can simply assign the promise back onto the i18n object, so that you can await that:

**next-i18next source code:**

```jsx
config.use.forEach(x => i18n.use(x))
const isInitializedPromise = i18n.init(config)
i18n.isInitializedPromise = isInitializedPromise
```

**User's code:**

```jsx
const server = express()
await nextI18next.i18n.isInitializedPromise
server.use(nextI18NextMiddleware(nextI18next))
```

There's also a possible argument that the init promise should be on the i18n object by default... Could get @jamuhl's opinion on that.

Let me know if this makes sense.

**Comment by jamuhl** [comment]
@isaachinman no reason I know of...so this looks ok to me.

**Comment by isaachinman** [comment]
@jamuhl So you'd accept a PR to put that promise on the object itself?

**Comment by jamuhl** [comment]
@isaachinman 

```js
config.use.forEach(x => i18n.use(x))
const isInitializedPromise = i18n.init(config)
i18n.isInitializedPromise = isInitializedPromise
```

that looks like code in next-i18next - not sure what you like to put on i18next?

if it's, adding the Promise inside init to `this`, pseudo code:

```js
init: (options) => {
   //...
   const deferred = defer();
   this.isInitializedPromise = deferred;
   // ...
}
```

would be ok...just that Promise will be overridden on each init call! or is this only about adding it to the typings?

**Comment by carloscuesta** [comment]
@isaachinman 

Your example looks right and is easy to understand follow and implement 👍 

Awaiting the initialisation of `i18n` ensures that the translations are loaded ? 🤔

**Comment by isaachinman** [comment]
> will be overridden on each init call

Good point. We'll keep this logic in the `next-i18next` source then.

@carloscuesta You can read the documentation [here](https://www.i18next.com/overview/api#init). Please feel free to submit a PR with this functionality. Must include at least basic test coverage. Thanks!

**Comment by isaachinman** [comment]
Any update?

**Comment by isaachinman** [comment]
Fixed by #616.
